File: Accenture/16.Find The Winner/code.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        arr = [i for i in range(1, n + 1)]
        i = 0 
        while len(arr) > 1:
            idx = (i + k - 1) % len(arr)
            arr.pop(idx)
            i = idx
        return arr[0]

File: Accenture/16.Find The Winner/code.cpp
#include <vector>
using namespace std;
class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> arr;
        for (int i = 1; i <= n; i++) {
            arr.push_back(i);  
        }
        int i = 0;  

        while (arr.size() > 1) {
            int idx = (i + k - 1) % arr.size();  
            arr.erase(arr.begin() + idx); 
            i = idx;  
        }
        return arr[0];  
    }
};
